Mathlib/Data/Set/Intervals/UnorderedInterval.lean

```diff
@@ -0,0 +1,390 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+
+! This file was ported from Lean 3 source module data.set.intervals.unordered_interval
+! leanprover-community/mathlib commit a59dad53320b73ef180174aae867addd707ef00e
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Order.Bounds.Basic
+import Mathlib.Data.Set.Intervals.Basic
+import Mathlib.Tactic.ScopedNS
+
+/-!
+# Intervals without endpoints ordering
+
+In any lattice `α`, we define `interval a b` to be `Icc (a ⊓ b) (a ⊔ b)`, which in a linear order is
+the set of elements lying between `a` and `b`.
+
+`Icc a b` requires the assumption `a ≤ b` to be meaningful, which is sometimes inconvenient. The
+interval as defined in this file is always the set of things lying between `a` and `b`, regardless
+of the relative order of `a` and `b`.
+
+For real numbers, `interval a b` is the same as `segment ℝ a b`.
+
+In a product or pi type, `interval a b` is the smallest box containing `a` and `b`. For example,
+`interval (1, -1) (-1, 1) = Icc (-1, -1) (1, 1)` is the square of vertices `(1, -1)`, `(-1, -1)`,
+`(-1, 1)`, `(1, 1)`.
+
+In `finset α` (seen as a hypercube of dimension `fintype.card α`), `interval a b` is the smallest
+subcube containing both `a` and `b`.
+
+## Notation
+
+We use the localized notation `[[a, b]]` for `interval a b`. One can open the locale `interval` to
+make the notation available.
+
+-/
+
+
+open Function
+
+open OrderDual (toDual ofDual)
+
+variable {α β : Type _}
+
+namespace Set
+
+section Lattice
+
+variable [Lattice α] {a a₁ a₂ b b₁ b₂ c x : α}
+
+/-- `interval a b` is the set of elements lying between `a` and `b`, with `a` and `b` included.
+Note that we define it more generally in a lattice as `set.Icc (a ⊓ b) (a ⊔ b)`. In a product type,
+`interval` corresponds to the bounding box of the two elements. -/
+def interval (a b : α) : Set α :=
+  Icc (a ⊓ b) (a ⊔ b)
+#align set.interval Set.interval
+
+-- Porting note: temporarily remove `scoped[Interval]` and use `[[]]` instead of `[]` before fix
```

Comment:
What fix? Is there a link to a discussion or PR?

Reply:
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/scoped.5BNS.5D/near/316207443

Perhaps fix isn’t the right word. I meant a workaround or solution.